lua/smart-splits/config.lua

```diff
@@ -8,24 +8,59 @@ local M = {
     'NvimTree',
   },
   move_cursor_same_row = false,
-  resize_mode_quit_key = '<ESC>',
-  resize_mode_silent = false,
+  resize_mode = {
+    quit_key = '<ESC>',
+    silent = false,
+    hooks = {
+      on_enter = nil,
+      on_leave = nil,
+    },
+  },
 }
 
 local function default_bool(value, default)
   if value == nil then
     return default
-  else
-    return value
   end
+  return value
+end
+
+local function default_hooks(new_config)
+  if not new_config then
+    return M.resize_mode.hooks
+  end
+  return {
+    on_enter = new_config.on_enter or M.resize_mode.hooks.on_enter,
+    on_leave = new_config.on_leave or M.resize_mode.hooks.on_leave,
+  }
+end
+
+local function default_resize_mode(new_config)
+  if not new_config then
+    return M.resize_mode
+  end
+  return {
+    quit_key = new_config.quit_key or M.resize_mode.quit_key,
+    silent = default_bool(new_config.silent, M.resize_mode.silent),
+    hooks = default_hooks(new_config.hooks),
+  }
 end
 
 function M.setup(config)
   M.ignored_buftypes = config.ignored_buftypes or M.ignored_buftypes
   M.ignored_filetypes = config.ignored_filetypes or M.ignored_filetypes
   M.move_cursor_same_row = default_bool(config.move_cursor_same_row, M.move_cursor_same_row)
-  M.resize_mode_quit_key = config.resize_mode_quit_key or M.resize_mode_quit_key
-  M.resize_mode_silent = default_bool(config.resize_mode_silent, M.resize_mode_silent)
+  M.resize_mode = default_resize_mode(config.resize_mode)
+
+  -- TODO: Remove this code block in the next commits
+  if config.resize_mode_quit_key then
+    M.resize_mode.quit_key = config.resize_mode_quit_key
+    vim.notify('smart-splits: resize_mode_quit_key has been changed to resize_mode.quit_key', vim.log.levels.WARN)
+  end
+  if config.resize_mode_silent then
+    M.resize_mode.silent = config.resize_mode_silent
+    vim.notify('smart-splits: resize_mode_silent has been changed to resize_mode.silent', vim.log.levels.WARN)
+  end
```

Comment:
thanks for adding this. can we update the message to specifically tell people they should update their config?

```suggestion
  -- TODO: Remove this code block in the next commits
  if config.resize_mode_quit_key then
    M.resize_mode.quit_key = config.resize_mode_quit_key
    vim.notify('smart-splits: resize_mode_quit_key has been changed to resize_mode.quit_key, please update your config. See README.md for details.', vim.log.levels.WARN)
  end
  if config.resize_mode_silent then
    M.resize_mode.silent = config.resize_mode_silent
    vim.notify('smart-splits: resize_mode_silent has been changed to resize_mode.silent, please update your config. See README.md for details.', vim.log.levels.WARN)
  end
```